src/qibo/backends/tensorflow.py

```diff
@@ -195,9 +196,7 @@ def __init__(self):
         elif cpu_devices:
             self.device = cpu_devices[0].name
 
-        import psutil
-
-        self.nthreads = psutil.cpu_count(logical=True)
+        self.nthreads = multiprocessing.cpu_count()
```

Comment:
Instead of guessing the number of available threads we might use `tf.config.get_inter_op_parallelism_threads` or `tf.config.get_intra_op_parallelism_threads` (please check which one is for single operator multi-threading) directly?

Reply:
```py
In [2]: tf.config.get_inter_op_parallelism_threads
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[2], line 1
----> 1 tf.config.get_inter_op_parallelism_threads

AttributeError: module 'tensorflow._api.v2.config' has no attribute 'get_inter_op_parallelism_threads'

In [3]: tf.config.get_intra_op_parallelism_threads
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[3], line 1
----> 1 tf.config.get_intra_op_parallelism_threads

AttributeError: module 'tensorflow._api.v2.config' has no attribute 'get_intra_op_parallelism_threads'
```

Reply:
https://www.tensorflow.org/api_docs/python/tf/config/threading/get_inter_op_parallelism_threads

Reply:
I believe they are now scoped in `tf.config.threading`:
https://www.tensorflow.org/api_docs/python/tf/config/threading/get_intra_op_parallelism_threads

Reply:
```py
In [10]:  tf.config.threading.get_inter_op_parallelism_threads()
Out[10]: 0

In [11]:  tf.config.threading.get_intra_op_parallelism_threads()
Out[11]: 0
```

And we do not set by default they are `0`:
> *A value of 0 means the system picks an appropriate number.*

Reply:
Exactly, so ideally you could check for 0, if True assume tf is using all threads given by psutil affinity.

Reply:
And what should I do if not zero?

1. set to zero myself
2. ignore it (I will assume the user chose it manually)
3. raise an error
4. check if it's the same of the affinity (and then do something)
5. set to the affinity

My favorite option is 2., since it conflicts the least with user choices (definitely a case in which less is more).
But in that case I should do nothing in general (not even check it if it's zero, since I will ignore anyhow the outcome)

Reply:
I think option 2 is reasonable, moreover `qibo.set_threads` consistently does not set threads when using the tf backend, but please double check if this value is not really needed anywhere.

Reply:
Thanks to @stavros11 I found it this interesting piece of Qibo:

https://github.com/qiboteam/qibo/blob/c1227d81b09a5e99f8be9e70a03cb7bcb41713b9/src/qibo/backends/tensorflow.py#L207-L214

So, I believe that it `qibo.set_threads` is not consistently doing its job. Especially for the `TensorflowBackend`. But I would not try to fix it in this PR.